pages/jobs/[slug].tsx

```diff
@@ -0,0 +1,64 @@
+import type { NextPage } from 'next';
+import type { JobPost } from 'lib/peopleHR';
+import styles from 'styles/Jobs.module.scss';
+
+const JobPosting: NextPage<{ job: JobPost }> = ({ job }) => {
+    return (
+        <div className={styles.container}>
+            <h1>{job.title}</h1>
+            <div
+                dangerouslySetInnerHTML={{
+                    __html: job.description,
+                }}
+            ></div>
+        </div>
+    );
+};
+
+export default JobPosting;
+
+const requestHeaders: HeadersInit = {
+    'Content-Type': 'application/json',
+    'Authorization': process.env.PEOPLEHR_AUTH_TOKEN as string,
+};
+
+export async function getStaticProps({ params }: { params: { slug: string } }) {
+    const apiURL =
+        process.env.NEXT_PUBLIC_FUNCTIONS_BASE_URL + '/api/jobs/' + params.slug;
+
+    try {
+        const job = await fetch(apiURL, { headers: requestHeaders }).then(
```

Comment:
I’m surprised the `fetch` API is available in Node just like that! Is that a Next.js thing?

Reply:
I am also surprised that this works. It may be a vercel thing, but it works when I run it using Next. Blog posts recommend using `node-fetch`  https://blog.maximeheckel.com/posts/data-fetching-with-next-js/